test/test_databases/test_dbsync.py

```diff
@@ -9,332 +12,192 @@
 TEMP_SCHEMA = "parsons_test"
 
 
-# These tests interact directly with the Postgres database. In order to run, set the
-# env to LIVE_TEST='TRUE'.
-@unittest.skipIf(not os.environ.get("LIVE_TEST"), "Skipping because not running live test")
-class TestPostgresDBSync(unittest.TestCase):
+class TestDBSync(ABC, unittest.TestCase):
+    setup_sql: Optional[str] = None
+    teardown_sql: Optional[str] = None
+    temp_schema: Optional[str] = TEMP_SCHEMA
+    db: Type[DatabaseConnector]
+
+    @classmethod
+    def setUpClass(cls):
+        # Skip tests on this abstract base class
+        if cls is TestDBSync:
+            raise unittest.SkipTest("%s is an abstract base class" % cls.__name__)
+        else:
+            super(TestDBSync, cls).setUpClass()
+
     def setUp(self):
-        self.temp_schema = TEMP_SCHEMA
-        self.db = Postgres()
+        self.initialize_db_connections()
 
-        # Create a schema.
-        setup_sql = f"""
-                     DROP SCHEMA IF EXISTS {self.temp_schema} CASCADE;
-                     CREATE SCHEMA {self.temp_schema};
-                     """
-        self.db.query(setup_sql)
+        if self.setup_sql:
+            self.source_db.query(self.setup_sql)
+            self.destination_db.query(self.setup_sql)
 
         # Load dummy data to parsons tables
         self.table1 = Table.from_csv(f"{_dir}/test_data/sample_table_1.csv")
         self.table2 = Table.from_csv(f"{_dir}/test_data/sample_table_2.csv")
 
+        self.source_table = (
+            f"{self.temp_schema}.source_table" if self.temp_schema else "source_table"
+        )
+        self.destination_table = (
+            f"{self.temp_schema}.destination_table" if self.temp_schema else "destination_table"
+        )
+
         # Create source table
-        self.db.copy(self.table1, f"{self.temp_schema}.source")
+        self.source_db.copy(self.table1, self.source_table, if_exists="truncate")
+
+        self.set_up_db_sync()
 
+    def set_up_db_sync(self, **kwargs) -> None:
         # Create DB Sync object
-        self.db_sync = DBSync(self.db, self.db)
+        self.db_sync = DBSync(self.source_db, self.destination_db, **kwargs)
 
-    def tearDown(self):
-        # Drop the view, the table and the schema
+    def initialize_db_connections(self) -> None:
+        self.source_db = self.db()
+        self.destination_db = self.db()
 
-        teardown_sql = f"""
-                       DROP SCHEMA IF EXISTS {self.temp_schema} CASCADE;
-                       """
-        self.db.query(teardown_sql)
+    def tearDown(self):
+        if self.teardown_sql:
```

Comment:
I'm curious when you *wouldn't* want to tear down the sql during a test? Making this conditional and not the default (since `teardown_sql` defaults to None) makes it likely that it will be run unintentionally - is the idea that it's better to accidentally not clean up after a test when you should than vice versa?  I don't do a ton of testing of databases, so I'm maybe missing something basic.

Reply:
Okay nevermind, I see that the sql is only getting run in the live tests where setup_sql and teardown_sql are set as attributes.

Reply:
Ya, there are circumstances where it's not necessary to define SQL to run on setup & teardown, so that's why the conditional. e.g. the SQLite connector creates an entirely new database on setup, and doesn't have a concept of schemas, so there's no need for SQL to run on setup or teardown.